Code/GraphMol/Chirality.cpp

```diff
@@ -429,52 +429,34 @@ bool isLinearArrangement(const RDGeom::Point3D &v1, const RDGeom::Point3D &v2) {
   return dotProd < cos178 * sqrt(lsq);
 }
 
-void updateDoubleBondNeighbors(ROMol &mol, Bond *dblBond, const Conformer *conf,
-                               boost::dynamic_bitset<> &needsDir,
-                               std::vector<unsigned int> &singleBondCounts,
-                               const VECT_INT_VECT &singleBondNbrs) {
-  // we want to deal only with double bonds:
-  PRECONDITION(dblBond, "bad bond");
-  PRECONDITION(dblBond->getBondType() == Bond::DOUBLE, "not a double bond");
-  if (!needsDir[dblBond->getIdx()]) {
-    return;
-  }
-  needsDir.set(dblBond->getIdx(), 0);
-#if 0
-  std::cerr << "**********************\n";
-  std::cerr << "**********************\n";
-  std::cerr << "**********************\n";
-  std::cerr << "UDBN: " << dblBond->getIdx() << " "
-            << dblBond->getBeginAtomIdx() << "=" << dblBond->getEndAtomIdx()
-            << "\n";
-#endif
-
-  std::vector<Bond *> followupBonds;
-
-  Bond *bond1 = nullptr, *obond1 = nullptr;
-  bool squiggleBondSeen = false;
-  bool doubleBondSeen = false;
-  for (const auto tBond : mol.atomBonds(dblBond->getBeginAtom())) {
+void controllingBondFromAtom(ROMol &mol, boost::dynamic_bitset<> &needsDir,
+                             std::vector<unsigned int> &singleBondCounts,
+                             Bond *dblBond, Atom *atom, Bond *&bond,
+                             Bond *&obond, bool &squiggleBondSeen,
+                             bool &doubleBondSeen) {
```

Comment:
I think `mol`, `needsDir`, `singleBondCounts`, `dblBond` and `atom` could be `const`.